core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -575,15 +583,32 @@ private[spark] object PythonRDD extends Logging {
     dataOut.write(bytes)
   }
 
-  def writeToFile[T](items: java.util.Iterator[T], filename: String) {
-    import scala.collection.JavaConverters._
-    writeToFile(items.asScala, filename)
-  }
+  private def serveIterator[T](items: Iterator[T]): Int = {
+    val serverSocket = new ServerSocket(0, 1)
+    serverSocket.setReuseAddress(true)
+    serverSocket.setSoTimeout(3000)
+
+    new Thread("serve iterator") {
+      setDaemon(true)
+      override def run() {
+        try {
+          val sock = serverSocket.accept()
+          val out = new DataOutputStream(new BufferedOutputStream(sock.getOutputStream))
+          try {
+            writeIteratorToStream(items, out)
+          } finally {
+            out.close()
+          }
+        } catch {
+          case NonFatal(e) =>
+            logError(s"Error while sending iterator: $e")
```

Comment:
If we change this to

```
logError("Error while sending iterator", e)
```

then the full stacktrace will be printed as part of the log message.